pandas/core/indexes/numeric.py

```diff
@@ -57,7 +57,7 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=None):
                 return cls._simple_new(data, name=name)
 
         # is_scalar, generators handled in coerce_to_ndarray
-        data = cls._coerce_to_ndarray(data)
+        data = cls._coerce_to_ndarray(data, dtype=dtype)
```

Comment:
I don't think this should rely on `dtype` being explicitly passed in (it's `None` by default) so maybe make this `dtype=self._dtype` for now since the `dtype` parameter isn't being validated.

Reply:
This initially made sense to me.

However, `numpy` seems to complicate this suggestion since it'll cast input if `dtype` is explicitly passed:

```python
In [1]: import numpy as np 
   ...: np.__version__                                                                                                                                                                                                                                                                                                         
Out[1]: '1.17.3'

In [2]: np.array([1, "2"]) 
   ...:                                                                                                                                                                                                                                                                                                                        
Out[2]: array(['1', '2'], dtype='<U21')

In [3]: np.array(["1", "2"])  
   ...:                                                                                                                                                                                                                                                                                                                        
Out[3]: array(['1', '2'], dtype='<U1')

In [4]: np.array(["1", "2"], dtype='int')                                                                                                                                                                                                                                                                                      
Out[4]: array([1, 2])
```

[This test](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexes/test_numeric.py#L565-L567) suggests that `pandas` keeps the same behavior. With [458c25e](https://github.com/pandas-dev/pandas/pull/29529/commits/458c25e89d8f44ce61e01e58796471b265436276#diff-248241e14f081931cc805f9d0e137a5fR4049-R4051), I tried to check the type of elements. The side effect of that is consuming iterators, leading to the current errors.

Reply:
The `numpy` behavior I report above is discussed [here](https://github.com/numpy/numpy/issues/10614#issuecomment-366386612) but not acted on.